src/app/_components/Form/FormFileInput.tsx

```diff
@@ -0,0 +1,179 @@
+import NextImage from 'next/image'
+
+import {
+  Field,
+  Input,
+  Description,
+  type InputProps as HeadlessInputProps,
+} from '@headlessui/react'
+import { Image, X } from '@phosphor-icons/react/dist/ssr'
+import { clsx } from 'clsx'
+
+import { FormError, type FormErrorProps } from '@/components/Form/FormError'
+import { FormLabel, type FormLabelProps } from '@/components/Form/FormLabel'
+import { Icon } from '@/components/Icon'
+
+type HardCodedProps = 'type' | 'onChange' | 'accept'
+
+export type FormFileInputProps = {
+  files: FileList | null
+  accept: Array<`.${string}`>
+  maxSize: number
+  description?: string | React.ReactNode
+  onChange: (files: FileList | null) => void
+} & Omit<HeadlessInputProps, HardCodedProps> &
+  FormLabelProps &
+  FormErrorProps
+
+export function FormFileInput({
+  files,
+  label,
+  hideLabel,
+  error,
+  hideError,
+  disabled,
+  onChange,
+  description,
+  maxSize,
+  accept,
+  ...rest
+}: FormFileInputProps) {
+  const loadedFile = files?.[0]
+
+  function resetFiles() {
+    onChange(null)
+  }
+
+  function loadFiles(e: React.ChangeEvent<HTMLInputElement>) {
+    const files = e.target.files
+
+    if (files) {
+      onChange(files)
+    }
+  }
+
+  return (
+    <Field className="w-full">
+      <FormLabel label={label} hideLabel={hideLabel} />
+
+      <div className="group relative h-48 w-full">
+        {loadedFile ? (
+          <SelectedFile file={loadedFile} onReset={resetFiles} />
+        ) : (
+          <>
+            <UploadInstructions
+              accept={accept}
+              maxSize={maxSize}
+              error={error}
+            />
+            <Input
+              {...rest}
+              type="file"
+              disabled={disabled}
+              accept={accept.join(',')}
+              invalid={Boolean(error)}
+              className={clsx(
+                'absolute inset-0 z-10 opacity-0',
+                disabled ? 'cursor-not-allowed' : 'cursor-pointer',
+              )}
+              onChange={loadFiles}
+            />
+          </>
+        )}
+      </div>
+
+      {description && (
+        <Description className="mt-4 text-sm text-brand-100">
+          {description}
+        </Description>
+      )}
+
+      <FormError error={error} hideError={hideError} />
+    </Field>
+  )
+}
+
+type SelectedFileProps = {
+  file: File
+  onReset: () => void
+}
+
+function SelectedFile({ file, onReset }: SelectedFileProps) {
+  return (
+    <div className="h-full w-full">
+      <NextImage
+        fill
+        src={URL.createObjectURL(file)}
+        alt={file.name}
+        className="rounded-lg object-cover"
+      />
```

Comment:
If it's a wide photo it might `object-cover` might cut it... `object-contain`? We use that in ecosystem projects

Reply:
As discussed with Filipa, we need `object-cover` so that the image uses the full width and the close button is always above it.